service/src/main/java/uk/gov/hmcts/reform/fpl/service/HearingBookingService.java

```diff
@@ -119,6 +151,99 @@ public HearingBooking getHearingBookingByUUID(List<Element<HearingBooking>> hear
             }).collect(toList());
     }
 
+    public List<Element<HearingBooking>> getNewHearings(
+        List<Element<HearingBooking>> newHearings, List<Element<HearingBooking>> oldHearings) {
+        List<UUID> oldHearingIDs = oldHearings.stream()
+            .map(Element::getId)
+            .collect(Collectors.toList());
+
+        return newHearings.stream()
+            .filter(newHearing -> !oldHearingIDs.contains(newHearing.getId()))
+            .collect(Collectors.toList());
+    }
+
+    public Map<String, Object> getHearingNoticeCaseFields(
+        List<Element<HearingBooking>> newHearings, List<Element<HearingBooking>> oldHearings) {
+        StringBuilder stringBuilder = new StringBuilder();
+        List<UUID> oldHearingIDs = oldHearings.stream()
+            .map(Element::getId)
+            .collect(Collectors.toList());
+
+        int selectorMinValue = -1;
+        for (int i = 0; i < newHearings.size(); i++) {
+            if (!oldHearingIDs.contains(newHearings.get(i).getId())) {
+                Element<HearingBooking> hearingElement = newHearings.get(i);
+                String newHearingLabel = format("Hearing %d: %s hearing %s", i + 1,
+                    hearingElement.getValue().getType().getLabel(),
+                    formatLocalDateTimeBaseUsingFormat(hearingElement.getValue().getStartDate(), DATE));
+
+                stringBuilder.append(newHearingLabel).append("\n");
+
+                if (selectorMinValue < 0) {
+                    selectorMinValue = i;
+                }
+            }
+        }
+        Selector selector = Selector.newSelector(newHearings.size(), selectorMinValue, newHearings.size());
+
+        return Map.of(NEW_HEARING_LABEL.getKey(), stringBuilder.toString(), NEW_HEARING_SELECTOR.getKey(), selector);
+    }
+
+    public List<Element<HearingBooking>> getSelectedHearings(Selector hearingSelector,
+                                                             List<Element<HearingBooking>> hearings) {
+        if (hearings.isEmpty() || hearingSelector == null) {
+            return List.of();
+        } else {
+            return hearingSelector.getSelected().stream()
+                .map(hearings::get)
+                .collect(toList());
+        }
+    }
+
+    public List<Element<HearingBooking>> getHearingDetails(
```

Comment:
As commented in controller I don't think "getHearingDetails" gives this method credit. What hearing details, maybe comment is required with this logic.